src/main/scala/com/databricks/spark/sql/perf/mllib/feature/VectorAssembler.scala

```diff
@@ -0,0 +1,47 @@
+package com.databricks.spark.sql.perf.mllib.feature
+
+import org.apache.spark.ml
+import org.apache.spark.ml.linalg.Vector
+import org.apache.spark.ml.PipelineStage
+import org.apache.spark.sql._
+import com.databricks.spark.sql.perf.mllib.OptionImplicits._
+import com.databricks.spark.sql.perf.mllib.data.DataGenerator
+import com.databricks.spark.sql.perf.mllib.{BenchmarkAlgorithm, MLBenchContext, TestFromTraining}
+
+/** Object for testing VectorAssembler performance */
+object VectorAssembler extends BenchmarkAlgorithm with TestFromTraining {
+
+  override def trainingDataSet(ctx: MLBenchContext): DataFrame = {
+    import ctx.params._
+    import ctx.sqlContext.implicits._
+
+    var df: DataFrame = null
+
+    for (i <- 1 to numInputCols.get) {
```

Comment:
for `Array.fill[T](numElem)(T)`, remove `.get` in `Array.fill(numFeatures)(featureArity.get)` I am afraid to cause ambiguous.